packages/aws-rfdk/lib/lambdas/nodejs/lib/ecr-provider/ecr-provider.ts

```diff
@@ -0,0 +1,186 @@
+   */
+  readonly products: {
+    /**
+     * Deadline images
+     */
+    readonly deadline: {
+      /**
+       * The URI namespace appended to the AWS Thinkbox Docker registry URI where
+       * Docker images can be pulled.
+       *
+       * Image Docker URIs can be computed as:
+       *
+       *     <REGISTRY_URI><DEADLINE_NAMESPACE><RECIPE_NAME>
+       */
+      readonly namespace: string;
+    };
+  };
+}
+
+/**
+ * The version provider parses a JSON file containing references to ECRs that serve Thinkbox's Deadline Docker images.
+ * It can be downloaded or loaded from local file and returns the ECR ARN prefix.
+ */
+export class ThinkboxEcrProvider {
+  /**
+   * The URL to obtain the ECR index from.
+   */
+  private static readonly ECR_INDEX_URL = 'https://downloads.thinkboxsoftware.com/thinkbox_ecr.json';
+
+  private indexJsonPromise?: Promise<ThinkboxDockerImageIndex>;
+
+  constructor(private readonly indexPath?: string) {}
+
+  private get indexJson() {
+    if (!this.indexJsonPromise) {
+      this.indexJsonPromise = new Promise<string>((resolve, reject) => {
+        try {
+          if (this.indexPath) {
+            return resolve(this.readEcrIndex(this.indexPath));
+          }
+          else {
+            // return resolve(this.getMockEcrIndex());
+            resolve(this.getEcrIndex());
+          }
+        }
+        catch (e) {
+          return reject(e);
+        }
+      }).then((json: string) => {
+        // convert the response to a json object and return.
+        let data: any;
+        try {
+          data = JSON.parse(json);
+        }
+        catch (e) {
+          throw new Error(`ECR index file contains invalid JSON: "${e}"`);
+        }
+
+        if (this.verifyThinkboxDockerImageIndex(data)) {
+          return data;
+        }
+        else {
+          throw new Error('This should be a dead code path');
+        }
+      });
+    }
+    return this.indexJsonPromise;
+  }
+
+  private verifyThinkboxDockerImageIndex(index: any): index is ThinkboxDockerImageIndex {
+    function expectObject(key: string, value: any) {
+      const valueType = typeof value;
+      if (valueType != 'object') {
+        throw new Error(`expected ${key} to be an object but got ${valueType}`);
+      }
+
+      if (Array.isArray(index)) {
+        throw new Error(`expected ${key} to be an object but got array`);
+      }
+    }
+
+    function expectString(key: string, value: any) {
+      const valueType = typeof value;
+      if (valueType != 'string') {
+        throw new Error(`expected ${key} to be a string but got ${valueType}`);
+      }
+    }
+
+    expectObject('index', index);
+    expectObject('index.registry', index.registry);
+    expectString('index.registry.uri', index.registry.uri);
+    expectObject('index.products', index.products);
+    expectObject('index.products.deadline', index.products.deadline);
+    expectObject('index.products.deadline', index.products.deadline);
```

Comment:
looks like this is duplicate and can be removed.

Reply:
Good catch. I've removed the duplicate now